xwiki-platform-core/xwiki-platform-resource/xwiki-platform-resource-servlet/src/main/java/org/xwiki/resource/servlet/AbstractServletResourceReferenceHandler.java

```diff
@@ -185,17 +187,35 @@ private void serveResource(R resourceReference, InputStream rawResourceStream)
 
     /**
      * Filter the resource before sending it to the client.
-     * 
+     *
      * @param resourceReference the resource to filter
      * @param resourceStream the resource content
      * @return the filtered resource content
+     * @deprecated use {@link #filterResource(ResourceReference, InputStream, HttpServletResponse)} instead
      */
+    @Deprecated
     protected InputStream filterResource(R resourceReference, InputStream resourceStream)
         throws ResourceReferenceHandlerException
     {
         return resourceStream;
     }
```

Comment:
I'll move this to legacy if we agree to add the `HttpServletResponse` parameter.

Reply:
Indeed, way cleaner that way, thanks for the suggestion.
In practice, `setResponseHeaders` is called before et content type definition, so setting the `Content-Type` here would have no effect since the value would be overridden by `response.setContentType`.
Instead I have introduced a protected `getContentType` method (see https://github.com/xwiki/xwiki-platform/pull/1586/files#diff-4ab936b179b6d22f41cb869da8f1e16018f3c43e9e8d3f3d19e8279e1ca0de5aR199)